downloader/src/main/kotlin/Downloader.kt

```diff
@@ -335,6 +340,12 @@ class Downloader {
                         log.warn { "Unable to delete temporary directory '$gemDirectory'." }
                     }
                 }
+            } else if (target.id.type == "Cargo") {
+                // without following redirect Cargo's package do not have an extension
+                val tarFile = File(
+                    sourceArchive.parentFile, sourceArchive.nameWithoutExtension + ".tgz")
```

Comment:
Above you say the name comes without an extension, so why not just use `.name` here to emphasize that again?

Reply:
I fixed this by using the filename from the last redirect and not the filename from the source archive.